<launch>
<!-- planning module -->
  <group ns="planning">
    <!-- lane_following scenario -->
    <group ns="scenario_planning/scenarios/lane_following">
      <!-- motion planning module -->
      <group ns="motion_planning">
        <arg name="input_path_topic" default="/planning/scenario_planning/scenarios/lane_following/behavior_planning/path" />
        <arg name="is_debug_clearance_map_mode" default="false" />
        <arg name="is_debug_drivable_area_mode" default="false" />
        <arg name="is_publishing_clearance_map_as_occupancy_grid" default="true" />
        <arg name="forward_fixing_distance" default="20" />
        <arg name="backward_fixing_distance" default="10" />
        <arg name="detecting_objects_radius_from_ego" default="50" />
        <arg name="detecting_objects_radius_around_path_point" default="2.5" />
        <arg name="number_of_backward_path_points_for_detecting_objects" default="5" />
        <arg name="exploring_minumum_radius" default="1.3" />
        <arg name="delta_arc_length_for_path_smoothing" default="1.0" />
        <arg name="delta_arc_length_for_explored_points" default="0.7" />
        <arg name="max_avoiding_objects_velocity_ms" default="1.1" />
        <arg name="clearance_weight_when_exploring" default="0.0" />
        <arg name="exploring_goal_clearance_from_obstacle" default="3.9" />
        <arg name="min_distance_threshold_when_switching_avoindance_to_path_following" default="0.1" />
        <arg name="min_cos_similarity_when_switching_avoindance_to_path_following" default="0.95" />
        <node pkg="path_planner" type="eb_path_planner_node" 
          name="eb_path_planner_node" output="screen">
          <remap from="~input/path" to="$(arg input_path_topic)" />
          <remap from="~output/path" to="path_planner/trajectory" />
          <param name="is_debug_clearance_map_mode" value="$(arg is_debug_clearance_map_mode)"/>
          <param name="is_debug_drivable_area_mode" value="$(arg is_debug_drivable_area_mode)"/>
          <param name="is_publishing_clearance_map_as_occupancy_grid" value="$(arg is_publishing_clearance_map_as_occupancy_grid)"/>
          <param name="forward_fixing_distance" value="$(arg forward_fixing_distance)"/>
          <param name="backward_fixing_distance" value="$(arg backward_fixing_distance)"/>
          <param name="detecting_objects_radius_from_ego" value="$(arg detecting_objects_radius_from_ego)"/>
          <param name="detecting_objects_radius_around_path_point" value="$(arg detecting_objects_radius_around_path_point)"/>
          <param name="number_of_backward_path_points_for_detecting_objects" value="$(arg number_of_backward_path_points_for_detecting_objects)"/>
          <param name="exploring_minumum_radius" value="$(arg exploring_minumum_radius)"/>
          <param name="delta_arc_length_for_path_smoothing" value="$(arg delta_arc_length_for_path_smoothing)"/>
          <param name="delta_arc_length_for_explored_points" value="$(arg delta_arc_length_for_explored_points)"/>
          <param name="max_avoiding_objects_velocity_ms" value="$(arg max_avoiding_objects_velocity_ms)"/>
          <param name="clearance_weight_when_exploring" value="$(arg clearance_weight_when_exploring)"/>
          <param name="exploring_goal_clearance_from_obstacle" value="$(arg exploring_goal_clearance_from_obstacle)"/>
          <param name="min_distance_threshold_when_switching_avoindance_to_path_following" value="$(arg min_distance_threshold_when_switching_avoindance_to_path_following)"/>
          <param name="min_cos_similarity_when_switching_avoindance_to_path_following" value="$(arg min_cos_similarity_when_switching_avoindance_to_path_following)"/>
        </node>
      </group> 
    </group>
  </group>
</launch>
